bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java

```diff
@@ -54,6 +53,20 @@
  */
 @NonNullByDefault
 public class NetworkUtils {
+
+    /**
+     * Nanos per millisecond.
+     */
+    private static final long NANOS_PER_MILLI = 1000_000L;
+
+    public static double durationToMillis(Duration duration) {
+        return (double) duration.toNanos() / NANOS_PER_MILLI;
```

Comment:
```suggestion
        return (double) duration.toMillis();
```

Reply:
You'd loose the precision see the JavaDoc:

> If this duration has greater than millisecond precision, then the conversion will drop any excess precision information as though the amount in nanoseconds was subject to integer division by one million.

The output of ping is usually more accurate:

```
PING www.google.com (172.217.23.196) 56(84) bytes of data.
64 bytes from prg03s05-in-f4.1e100.net (172.217.23.196): icmp_seq=1 ttl=59 time=4.68 ms
64 bytes from prg03s05-in-f196.1e100.net (172.217.23.196): icmp_seq=2 ttl=59 time=11.1 ms
```

Reply:
Consider adding a comment to reduce the risk of someone "optimizing" this in the future.

Reply:
OK I'll add a comment! There are also unit tests to prevent such regressions. :wink:

Reply:
In that case I rest my case. 😀